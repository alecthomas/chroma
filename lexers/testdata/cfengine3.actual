@if minimum_version(3.20)
body members foo
{
  include => { "alice", "bob" };
  exclude => { "malcom" };
}
@endif

@if minimum_version(3.20)
promise agent blah
{
  path => "/some/path";
}
@endif

bundle agent __main__
{
  groups:
    "foo"
      policy => "present",
@if minimum_version(3.20)
      members => foo;
@else
      members => '{ "include": ["alice", "bob"],
                    "exclude": ["malcom"] }';
@endif
}
