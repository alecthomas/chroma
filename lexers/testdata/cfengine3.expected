[
  {"type":"CommentPreproc","value":"@if minimum_version(3.20)\n"},
  {"type":"Keyword","value":"body"},
  {"type":"Text","value":" "},
  {"type":"Keyword","value":"members"},
  {"type":"Text","value":" "},
  {"type":"NameFunction","value":"foo"},
  {"type":"Text","value":"\n"},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n  "},
  {"type":"KeywordReserved","value":"include"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"=\u003e"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":" "},
  {"type":"LiteralString","value":"\"alice\""},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"LiteralString","value":"\"bob\""},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"};"},
  {"type":"Text","value":"\n  "},
  {"type":"KeywordReserved","value":"exclude"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"=\u003e"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":" "},
  {"type":"LiteralString","value":"\"malcom\""},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"};"},
  {"type":"Text","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n"},
  {"type":"CommentPreproc","value":"@endif\n"},
  {"type":"Text","value":"\n"},
  {"type":"CommentPreproc","value":"@if minimum_version(3.20)\n"},
  {"type":"Keyword","value":"promise"},
  {"type":"Text","value":" "},
  {"type":"Keyword","value":"agent"},
  {"type":"Text","value":" "},
  {"type":"NameFunction","value":"blah"},
  {"type":"Text","value":"\n"},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n  "},
  {"type":"KeywordReserved","value":"path"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"=\u003e"},
  {"type":"Text","value":" "},
  {"type":"LiteralString","value":"\"/some/path\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n"},
  {"type":"CommentPreproc","value":"@endif\n"},
  {"type":"Text","value":"\n"},
  {"type":"Keyword","value":"bundle"},
  {"type":"Text","value":" "},
  {"type":"Keyword","value":"agent"},
  {"type":"Text","value":" "},
  {"type":"NameFunction","value":"__main__"},
  {"type":"Text","value":"\n"},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n  "},
  {"type":"KeywordDeclaration","value":"groups"},
  {"type":"Punctuation","value":":"},
  {"type":"Text","value":"\n    "},
  {"type":"LiteralString","value":"\"foo\""},
  {"type":"Text","value":"\n      "},
  {"type":"KeywordReserved","value":"policy"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"=\u003e"},
  {"type":"Text","value":" "},
  {"type":"LiteralString","value":"\"present\""},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":"\n"},
  {"type":"CommentPreproc","value":"@if minimum_version(3.20)\n"},
  {"type":"Text","value":"      "},
  {"type":"KeywordReserved","value":"members"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"=\u003e"},
  {"type":"Text","value":" "},
  {"type":"NameFunction","value":"foo"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"CommentPreproc","value":"@else\n"},
  {"type":"Text","value":"      "},
  {"type":"KeywordReserved","value":"members"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"=\u003e"},
  {"type":"Text","value":" "},
  {"type":"LiteralString","value":"'{ \"include\": [\"alice\", \"bob\"],\n                    \"exclude\": [\"malcom\"] }'"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"CommentPreproc","value":"@endif\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n"}
]
